kroxylicious-authorizer-providers/kroxylicious-authorizer-acl/src/main/java/io/kroxylicious/authorizer/provider/acl/AclAuthorizerService.java

```
@@ -0,0 +1,463 @@
+            if (principalClass != null) {
+                this.principalBuilder = this.subjectBuilder.subjectsHavingPrincipal(principalClass);
+            }
+            this.subjectBuilder = null;
+        }
+
+        private <T> @Nullable Class<? extends T> lookupClass(TerminalNode node, Class<T> cls, String desc) {
+            // look it up in the imports
+            String localClassName = node.getText();
+            var qualifiedClassName = this.localToQualified.get(localClassName);
+            if (qualifiedClassName == null) {
+                reportError(node.getSymbol(),
+                        "%s class with name '%s' has not been imported.".formatted(desc, localClassName));
+                return null;
+            }
+            try {
+                Class<?> c = Class.forName(qualifiedClassName);
+                if (!cls.isAssignableFrom(c)) {
+                    reportError(node.getSymbol(),
+                            "%s class '%s' is not a subclass of %s.".formatted(desc, localClassName, cls));
+                    return null;
+                }
+                return c.asSubclass(cls);
+            }
+            catch (ClassNotFoundException e) {
+                reportError(node.getSymbol(),
+                        "%s class '%s' was not found.".formatted(desc, qualifiedClassName));
+                return null;
+            }
+        }
+
+        @Override
+        public void enterNameEq(AclRulesParser.NameEqContext ctx) {
+            if (this.resourceBuilder != null) {
+                this.resourceBuilder.onResourceWithNameEqualTo(unquoteString(ctx.STRING()));
+                this.resourceBuilder = null;
+            }
+            else if (this.principalBuilder != null) {
+                this.operationsBuilder = this.principalBuilder.withNameEqualTo(unquoteString(ctx.STRING()));
+                this.principalBuilder = null;
+            }
```

Should there be an else clause here? Either logging or failing. 

I'm worried about silent failures